libs/SmartSync/src/com/salesforce/androidsdk/smartsync/manager/SyncManager.java

```diff
@@ -255,7 +255,8 @@ public void run() {
                     updateSync(sync, SyncState.Status.DONE, 100, callback);
                 } catch (RestClient.RefreshTokenRevokedException re) {
                     SmartSyncLogger.e(TAG, "Exception thrown in runSync", re);
-                    // Do not do anything - let the logout go through!
+                    // Failed due to invalid refresh token. Update status to failed.
+                    updateSync(sync, SyncState.Status.FAILED, UNCHANGED, callback);
```

Comment:
We didn't do an updateSync on purpose because it will try and write to the database, but the revoke token will cause a logout which will cause the database to be wiped out.
Maybe we can have updateSync take an extra argument to decide whether to just notify listeners or notify listeners and update status in db. We would use the only notify listeners in this case?